spacelift/resource_bitbucket_datacenter_integration.go

```diff
@@ -0,0 +1,213 @@
+	}
+
+	fillResults(d, &mutation.CreateBitbucketDatacenterIntegration)
+
+	return nil
+}
+
+func resourceBitbucketDatacenterIntegrationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var query struct {
+		BitbucketDatacenterIntegration *structs.BitbucketDatacenterIntegration `graphql:"bitbucketDatacenterIntegration(id: $id)"`
+	}
+
+	variables := map[string]interface{}{"id": d.Id()}
+	if err := meta.(*internal.Client).Query(ctx, "BitbucketDatacenterIntegrationRead", &query, variables); err != nil {
+		return diag.Errorf("could not query for the bitbucket datacenter integration: %v", err)
+	}
+
+	if query.BitbucketDatacenterIntegration == nil {
+		d.SetId("")
+	} else {
+		fillResults(d, query.BitbucketDatacenterIntegration)
+	}
+
+	return nil
+}
+
+func resourceBitbucketDatacenterIntegrationUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var mutation struct {
+		UpdateBitbucketDatacenterIntegration structs.BitbucketDatacenterIntegration `graphql:"bitbucketDatacenterIntegrationUpdate(apiHost: $apiHost, userFacingHost: $userFacingHost, username: $username, accessToken: $accessToken, customInput: $customInput)"`
+	}
+
+	variables := map[string]interface{}{
+		"apiHost":        toString(d.Get(bitbucketDatacenterAPIHost)),
+		"userFacingHost": toString(d.Get(bitbucketDatacenterUserFacingHost)),
+		"username":       toOptionalString(d.Get(bitbucketDatacenterUsername)),
+		"accessToken":    toOptionalString(d.Get(bitbucketDatacenterAccessToken)),
+		"customInput": &structs.CustomVCSUpdateInput{
+			ID:          toID(d.Id()),
+			SpaceID:     toString(d.Get(bitbucketDatacenterSpaceID)),
+			Description: toOptionalString(d.Get(bitbucketDatacenterDescription)),
+			Labels:      toOptionalStringList(d.Get(bitbucketDatacenterLabels)),
+		},
+	}
+
+	var ret diag.Diagnostics
+
+	if err := meta.(*internal.Client).Mutate(ctx, "BitbucketDatacenterIntegrationUpdate", &mutation, variables); err != nil {
+		ret = append(ret, diag.Errorf("could not update the bitbucket datacenter integration: %v", internal.FromSpaceliftError(err))...)
+	}
+
+	fillResults(d, &mutation.UpdateBitbucketDatacenterIntegration)
+
+	return ret
+}
+
+func resourceBitbucketDatacenterIntegrationDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var mutation struct {
+		DeleteBitbucketDatacenterIntegration *structs.BitbucketDatacenterIntegration `graphql:"bitbucketDatacenterIntegrationDelete(id: $id)"`
+	}
+
+	variables := map[string]interface{}{
+		"id": toID(d.Id()),
+	}
+
+	if err := meta.(*internal.Client).Mutate(ctx, "BitbucketDatacenterIntegrationDelete", &mutation, variables); err != nil {
+		return diag.Errorf("could not delete bitbucket datacenter integration: %v", internal.FromSpaceliftError(err))
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func fillResults(d *schema.ResourceData, bitbucketDatacenterIntegration *structs.BitbucketDatacenterIntegration) {
```

Comment:
I believe this is a package for all resources, therefore how about calling this function `fillBitbucketDatacenterIntegrationResults` as otherwise it's going to pop up with a confusing name whenever we develop other resources.

Reply:
Sure, it makes sense.